Promote various small API additions introduced in 6.x releases

jjohannes opened the pull request:
### Checklist
- [x] Validate whether we should de-incubate the API in the 7.0 release.
    - If the removal is not possible, create a follow-up issue and link it in the [overview spreadsheet](https://docs.google.com/spreadsheets/d/19J1nR_dFKpfKdu5KDFMVZGfjR0ysT9DthsBUPwf8mkM/edit#gid=1195622786)
- [x] Update release notes (add API to promoted features)
- [x] Check User Manual (it might mention that the API is still incubating)
  - Think about updating snippets and samples to use this API
- Deprecate existing API that is replaced by the new one: https://github.com/gradle/gradle/issues/15681
- [x] Check Incubation Report on TeamCity ([example](https://builds.gradle.org/viewLog.html?buildId=40024670&buildTypeId=Gradle_Check_SanityCheck&tab=report_project951_Incubating_APIs_Report)) 
- [ ] Mark the story as done in the [overview spreadsheet](https://docs.google.com/spreadsheets/d/19J1nR_dFKpfKdu5KDFMVZGfjR0ysT9DthsBUPwf8mkM/edit?ts=5fcfefb8#gid=0).

jjohannes commented:
@bot-gradle test QF

bot-gradle commented:
<!-- {"replyTargetCommentId":760961386,"teamCityBuildId":"40654008","teamCityBuildHeadRef":"e07942ef0152dc4a84d206735b4f10d5620f6319","teamCityBuildStatus":"PENDING","replyApproveReviewId":null} -->
OK, I've already triggered [QuickFeedback build](https://builds.gradle.org/viewQueued.html?itemId=40654008) for you.

jjohannes commented:
@bot-gradle test QF

bot-gradle commented:
<!-- {"replyTargetCommentId":760985365,"teamCityBuildId":"40655394","teamCityBuildHeadRef":"566937bca5259c9a18c252881696674ef0ed50f6","teamCityBuildStatus":"PENDING","replyApproveReviewId":null} -->
OK, I've already triggered [QuickFeedback build](https://builds.gradle.org/viewQueued.html?itemId=40655394) for you.

ljacomet reviewed (approved):
LGTM but there are CI failures which I did not understand with a quick look

jjohannes commented:
> CI failures which I did not understand with a quick look

I thought we can als add `ReplaceBy` to setters. To indicate to users that they should use a new property to set a value. So I added one. Apparently that is not allowed. Removed it again.

jjohannes commented:
@bot-gradle test QFL

bot-gradle commented:
<!-- {"replyTargetCommentId":761112734,"teamCityBuildId":"40661351","teamCityBuildHeadRef":"e1bfc7558354bdaa3d9917c318061da8d7d662b6","teamCityBuildStatus":"PENDING","replyApproveReviewId":null} -->
OK, I've already triggered [QuickFeedbackLinux build](https://builds.gradle.org/viewQueued.html?itemId=40661351) for you.
